Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -131,36 +141,67 @@ public class AWSClient {
         case .useAWSClientShared:
             self.eventLoopGroup = AWSClient.sharedEventLoopGroup
         }
+        self.httpClient = AWSClient.createHTTPClient(eventLoopGroup: self.eventLoopGroup)
```

Comment:
Given we prefix most of the values below with `self.` I would say that is consistent with the rest of the function. I think this is fine within an init() function.

Reply:
hmm, not exactly. The `self.` prefix is required on the assignment (left) side and we use that appropriately to assign values below. But it's not required when reading the value of the `eventLoopGroup` instance constant. I'm suggesting to remove `self.` from where we read that value.

So it would be

`self.httpClient = AWSClient.createHTTPClient(eventLoopGroup: eventLoopGroup)`

If that makes sense.

Then to disambiguate above when we are first assigning that constant, it would be better to name the unwrapped EventLoopGroup something that indicates it's not the local instance var. So for that I'm recommending `sharedEventLoopGroup`

Let me know if this still doesn't make sense. I can explain over discord.

And goes without saying - these are nit picks and not a big deal. Just things that I notice when I'm reading the code that imo would make it more clear what's happening.

Reply:
Sorry missed which bit you were talking about. Removed `self` and renamed `eventLoopGroup` in the switch statement